### PR TITLE
Feat/refactor

### PR DESCRIPTION
**Pull request by ahmedheakl**
Fix - Solve convex hull difficulty
###### Ticket(s): [run-the-code](https://app.clickup.com/t/8669g6bfr)

- 🔄 **Main changes in this PR**
    - Add option to view library version from argparser using `highrl -v` or `highrl --version`.
    - Refactored `env`, `utils`, `callbacks`, `agents` modules.
    - Added pylint and mypy to the CI.

- ⚠️ **Breaking changes**
  - Argument `coords` in `utils.planner_checker.get_region_coords` changed to `robot_goal_coords` for better readability.

## 🚀 How to test this Pull Request

  - Run `pytest test`
## 📝 Notes for future improvements

- Refactor `teacher_checker.py`

### CODE REVIEW

**Review by AbdelrahmanAbdelgwad — APPROVED** [review]
LGTM!!

File: src/highrl/envs/eval_env.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def _generate_obstacles_points(
                 new_height = randint(min_dim, max_dim)
                 new_obstacle = SingleObstacle(px, py, new_width, new_height)
                 overlap = self.robot.is_overlapped(
-                    new_obstacle, check_target="robot"
+                    new_obstacle, check_target="agent"
```
Comment (abdualamer) [inline src/highrl/envs/eval_env.py:93]: why did we change robot to agent  ?

Comment (AbdelrahmanAbdelgwad) [reply src/highrl/envs/eval_env.py:93]: More generic + it is in the agent.py file so makes more sense.

File: src/highrl/envs/teacher_env.py [hunk 2]
```diff
@@ -402,45 +507,43 @@ def __get_reward(self) -> float:
...
-            )
-            ** self.alpha
-            + (
-                self.terminal_state_flag
-                * (1 - self.robot_env.episode_steps / self.max_robot_episode_steps)
-                * self.terminal_state_reward
-            )
-            + (self.difficulty_area - self.desired_difficulty) * self.gamma
+        dfc_fact = self.difficulty_area / self.desired_difficulty
+        rwd_fact = float(self.robot_avg_reward / self.max_robot_episode_reward)
```
Comment (abdualamer) [inline src/highrl/envs/teacher_env.py:511]: I think these variables could be more descriptive

**Review by abdualamer — APPROVED** [review]
All good.

**Pull request merged**